3741/3/helpers/user_helper.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'sinatra/base'
+
+module UserHelper
+  def sign_up_redirect
+    if register_data_valid?
+      @user.save
+      show_message 'user create, login in please'
+      redirect '/'
+    else
+      show_message 'email or login already exist'
+      redirect '/session/signup'
+    end
+  end
+
+  def sign_in_redirect
+    if user_data_valid?
+      session[:user_id] = @user.id
+      redirect '/'
+    else
+      show_message 'wrong email'
+      redirect '/session/login'
+    end
+  end
+
+  def register_data_valid?
+    @user = User.new(params.slice('username', 'email', 'password'))
+    @user.valid? && Truemail.valid?(params['email'])
```

Comment:
`Truemail.valid?(params['email'])` this code looks like custom validation. To create one you just need to do
```ruby
validate :your_custom_validation_method_name
```

Reply:
i try, but dont understand why this not working in sinatra :(
![image](https://user-images.githubusercontent.com/36412453/62404226-4beea480-b59b-11e9-8a63-98c37e9b3cc9.png)

Reply:
It is working buy you need to add error to error object of model
```ruby
error.add(...) if Truemail.valid?(email)
```